### ISSUE TITLE
Problem calling const functions with dynamic array return type

### ISSUE THREAD

**Issue by Nashatyrev** [ethereum/web3.js#336]
I deployed a contract which contains the following functions:

``` js
    function getString() constant returns (string) {
        return "This is a very long string: longer than 32 bytes";
    }

    uint[] arr;
    function getArray() constant returns (uint[]) {
        arr.length = 2;
        arr[0] = 0x111111111111;
        arr[1] = 0x222222222222;
        return arr;
    }
```

Running the following script:

``` js
var myabi = 
[
  {
    "constant": true,
    "inputs": [],
    "name": "getString",
    "outputs": [
      {
        "name": "",
        "type": "string"
      }
    ],
    "type": "function"
  },
  {
    "constant": true,
    "inputs": [],
    "name": "getArray",
    "outputs": [
      {
        "name": "",
        "type": "uint256[]"
      }
    ],
    "type": "function"
  },
];

var ret = web3.eth.call({
    to: "0x8901a2bbf639bfd21a97004ba4d7ae2bd00b8da8",
    from: "0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826",
    data: "0x89ea642f"  // getString()
  });
console.log(ret);
var ret = web3.eth.call({
    to: "0x8901a2bbf639bfd21a97004ba4d7ae2bd00b8da8",
    from: "0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826",
    data: "0xd504ea1d"  // getArray()
  });
console.log(ret);

var contract = web3.eth.contract(myabi).at('0x8901a2bbf639bfd21a97004ba4d7ae2bd00b8da8');
var ret = contract.getArray({from: "0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826"});
console.log(ret);
var ret = contract.getString({from: "0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826"} );
console.log(ret);
```

Results in the following out:

```
0x2000000000000000000000000000000000000000000000000000000000000000305468697320697320612076657279206c6f6e6720737472696e673a206c6f6e676572207468616e20333220627974657300000000000000000000000000000000
0x20000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000001111111111110000000000000000000000000000000000000000000000000000222222222222
[]
/root/workspace/test/node_modules/web3/node_modules/bignumber.js/bignumber.js:1209
            throw error;
                  ^
BigNumber Error: new BigNumber() not a base 16 number: 
    at raise (/root/workspace/test/node_modules/web3/node_modules/bignumber.js/bignumber.js:1177:25)
    at /root/workspace/test/node_modules/web3/node_modules/bignumber.js/bignumber.js:1165:33
    at new BigNumber (/root/workspace/test/node_modules/web3/node_modules/bignumber.js/bignumber.js:212:28)
    at formatOutputString [as _outputFormatter] (/root/workspace/test/node_modules/web3/lib/solidity/formatters.js:217:19)
    at /root/workspace/test/node_modules/web3/lib/solidity/type.js:237:25
    at SolidityType.decode (/root/workspace/test/node_modules/web3/lib/solidity/type.js:238:11)
    at /root/workspace/test/node_modules/web3/lib/solidity/coder.js:219:29
    at Array.map (native)
    at SolidityCoder.decodeParams (/root/workspace/test/node_modules/web3/lib/solidity/coder.js:218:26)
    at SolidityFunction.unpackOutput (/root/workspace/test/node_modules/web3/lib/web3/function.js:90:24)
```

I.e. the contract returns correct binary data, but the web3 fails to parse it.

**Comment by aaronstjohn** [comment]
I'm having what I think is a related issue, in my case, web3 tries to convert a return value that is an array of 'ints' to a big number and fails in the process. You can see in the error below, that the array came through but got blocked by BigNumber.

The contract code I'm running looks like so. Its just pushing a test data-set to a mod-pow implementation and dumping out the results.

```
function PowArr(int[] base, int[] exp, int[] mod) returns (int[] output)
 {
    if(base.length != mod.length || base.length != exp.length ) throw;
    for(uint i=0 ; i< base.length; i++)
    {
      output[i] = Pow(base[i],exp[i],mod[i]);
    }
    return output;
 }
```

I'm calling web3 from truffle/pudding but this error triggers from web3. I double checked that the distro of pudding I'm using has the latest version of web3.


Here are the errors:
```
Unhandled rejection BigNumber Error: new BigNumber() not a number: 40083,40086,40088,40095,40103,40104,40110,40115,40116,40119
  at raise (/Projects/Ethereum/truffle/node_modules/web3/node_modules/bignumber.js/bignumber.js:1177:25)
  at /Projects/Ethereum/truffle/node_modules/web3/node_modules/bignumber.js/bignumber.js:1165:33
  at new BigNumber (/Projects/Ethereum/truffle/node_modules/web3/node_modules/bignumber.js/bignumber.js:193:67)
  at new BigNumber (/Projects/Ethereum/truffle/node_modules/web3/node_modules/bignumber.js/bignumber.js:203:25)
  at toBigNumber (/Projects/Ethereum/truffle/node_modules/web3/lib/utils/utils.js:362:12)
  at Object.toTwosComplement (/Projects/Ethereum/truffle/node_modules/web3/lib/utils/utils.js:373:21)
  at [object Object].formatInputInt [as _inputFormatter] (/Projects/Ethereum/truffle/node_modules/web3/lib/solidity/formatters.js:40:38)
  at [object Object].SolidityType.encode (/Projects/Ethereum/truffle/node_modules/web3/lib/solidity/type.js:179:17)
  at /Projects/Ethereum/truffle/node_modules/web3/lib/solidity/coder.js:86:29
  at Array.map (native)
  at [object Object].SolidityCoder.encodeParams (/Projects/Ethereum/truffle/node_modules/web3/lib/solidity/coder.js:85:34)
  at [object Object].SolidityFunction.toPayload (/Projects/Ethereum/truffle/node_modules/web3/lib/web3/function.js:69:52)
  at [object Object].SolidityFunction.call (/Projects/Ethereum/truffle/node_modules/web3/lib/web3/function.js:108:24)

  ...
```

**Comment by frozeman** [comment]
Can you test this again with the newly released 0.18.0 version?

**Comment by frozeman** [comment]
Please test with web3.js 1.0.0. Re-open if its still an issue
